store/RedisStore.js

```diff
@@ -1,50 +1,88 @@
 'use strict'
 
+const luaBasic = `
+  -- Key to operate on
+  local key = KEYS[1]
+  -- Time window for the TTL
+  local timeWindow = tonumber(ARGV[1])
+
+  -- Increment the key's value
+  local value = redis.call('INCR', key)
+
+  -- Check the current TTL of the key
+  local ttl = redis.call('PTTL', key)
+
+  -- If the key is new then set its TTL
+  if ttl == -1 then
+      redis.call('PEXPIRE', key, timeWindow)
+      return {1, timeWindow}
+  end
+
+  return {value, ttl}
+`
+
+const luaContinueExceeding = `
+  -- Key to operate on
+  local key = KEYS[1]
+  -- Time window for the TTL
+  local timeWindow = tonumber(ARGV[1])
+  -- Max allowed value
+  local max = tonumber(ARGV[2])
+
+  -- Increment the key's value
+  local value = redis.call('INCR', key)
+
+  -- Check the current TTL of the key
+  local ttl = redis.call('PTTL', key)
+
+  -- If the key is new then set its TTL
+  if ttl == -1 then
+      redis.call('PEXPIRE', key, timeWindow)
+      return {1, timeWindow}
+  -- If the key's incremented value has exceeded the max value, then reset its TTL
+  elseif value > max then
+      redis.call('PEXPIRE', key, timeWindow)
+      return {value, timeWindow}
+  end
```

Comment:
```suggestion
  elseif value > max then
      redis.call('PEXPIRE', key, timeWindow)
      return {value, timeWindow}
  end
```

If this part is the only difference, why don't we share the same script with another argument?

Reply:
You are right, this whole complexity is caused by having two scripts in the first place

I will update